plugins/woocommerce-admin/client/products/fields/category-field/category-field-add-new-item.tsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Icon } from '@wordpress/components';
+import { plus } from '@wordpress/icons';
+import classNames from 'classnames';
+import { ProductCategory } from '@woocommerce/data';
+import { __experimentalSelectControlMenuItemProps as MenuItemProps } from '@woocommerce/components';
+
+type CategoryFieldAddNewItemProps = {
+	item: Pick< ProductCategory, 'id' | 'name' >;
+	highlightedIndex: number;
+	items: Pick< ProductCategory, 'id' | 'name' >[];
+} & Pick<
+	MenuItemProps< Pick< ProductCategory, 'id' | 'name' > >,
+	'getItemProps'
+>;
+
+export const CategoryFieldAddNewItem: React.FC<
+	CategoryFieldAddNewItemProps
+> = ( { item, highlightedIndex, getItemProps, items } ) => {
+	const index = items.findIndex( ( i ) => i.id === item.id );
+	return (
+		<li
+			{ ...getItemProps( {
+				item,
+				index,
+			} ) }
+			className={ classNames(
+				'woocommerce-category-field-dropdown__item woocommerce-category-field-dropdown__new-item',
```

Comment:
Nitpicky, but this has 2 class names used as identifiers.  Should one of these be a modifier?  E.g., `woocommerce-category-field-dropdown__item is-new`